datafusion/core/tests/sqllogictests/test_files/decimal.slt

```diff
@@ -365,7 +365,7 @@ select c1*c5 from decimal_simple;
 query T
 select arrow_typeof(c1/cast(0.00001 as decimal(5,5))) from decimal_simple limit 1;
 ----
-Decimal128(21, 12)
+Decimal128(19, 10)
```

Comment:
cc @liukun4515 and @viirya -- I assume you have reviewed the decimal changes upstream

Reply:
I won't say this is not correct, this is more like a design decision to me. But this actually changes the precision of decimal arithmetic operations in DataFusion. For some, this might be a different change and not something could be implicitly accepted. I'd suggest there are some documents on these changes with new release including this.

Reply:
> But this actually changes the precision of decimal arithmetic operations in DataFusion

Correct, it changes the semantics of decimal division to match that of postgres, mysql and Hive. I've not done an exhaustive survey of other DBMS products, but the Hive semantics were decided based on a survey of the systems at the time. Other decimal operations are unaffected. This should be definitely communicated, especially if Spark does something different (possibly as a misreading of the Hive specification).

Reply:
I think spark is of the other major targets / comparison system -- I think @liukun4515  and maybe @yjshen  perhaps would have more context / understanding of how potential changes to the decimal division semantics may affect downstream users

If there is need for both semantics, maybe we could add a  "use_spark_decimal_division_rules" config option to DataFusion